### DISCUSSION TITLE
Always redirect from page A to page B

### DISCUSSION THREAD

**Discussion by sanderkooger** [vercel/next.js#13662] [Help] [answered]
All, 

I' m a new developer trying to hack together a Media on demand app. 

In this app i have the following page structure: 
```
src
   ├── pages
        ├── library.js <== Redirect to ./library/catwalks instantly
        └── index.js
                ├── library
                └── catwalks.js

```
I want to sort my library pages in a subdirectory, but i want the /library url to always show the catwalks page. 

How do i make this happen?

**Comment by jamesmosier** [comment, marked as answer]
One way to do it within Next would be to add this to your `pages/library.js` (or more likely  `pages/library/index.js`):

```js
export async function getServerSideProps(context) {
  context.res.writeHead(302, {
    Location: '/library/catwalks',
  });

  context.res.end();

  return {
    props: {}, 
  };
}
```

**Comment by timneutkens** [comment]
Opened a RFC here: https://github.com/vercel/next.js/discussions/14890
